meta/saisanitycheck.c

```diff
@@ -3960,7 +3960,9 @@ void check_acl_table_fields_and_acl_entry_fields()
          * check acl table flags and attr value type
          */
 
-        if (mtable->attrid == SAI_ACL_TABLE_ATTR_FIELD_ACL_RANGE_TYPE)
+        if ((mtable->attrid == SAI_ACL_TABLE_ATTR_FIELD_ACL_RANGE_TYPE) ||
+        ((mtable->attrid >= SAI_ACL_TABLE_ATTR_USER_DEFINED_FIELD_GROUP_MIN) &&
```

Comment:
why this condition is added? seems like not neeed, remove ?

Reply:
Attributes between SAI_ACL_TABLE_ATTR_FIELD_START and SAI_ACL_TABLE_ATTR_FIELD_END are expected to be bool.

RANGE_TYPE and USER_DEFINED_FIELD_GROUP are exception to this rule as the length can not be derived from these attributes directly.

......
Error message: If toolchain is not modified for exception

 ASSERT FAILED (on line 3980): SAI_ACL_TABLE_ATTR_USER_DEFINED_FIELD_GROUP_MIN: acl table attr value type should be bool

Reply:
why then those attributes are changed from bool to oid ?

Reply:
Kamil, more details are in the spec. There is no way to get length of the UDF field just by the bool, it is stored in the UDF group OID hence the change.

Reply:
@kcudnik, is it ok?

Reply:
Thanks